sdk/openai/azure-ai-openai/src/test/java/com/azure/ai/openai/OpenAIClientBuilderTest.java

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.ai.openai;
+
+import java.lang.reflect.Field;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import org.junit.jupiter.api.Test;
+import static com.azure.ai.openai.implementation.NonAzureOpenAIClientImpl.EU_OPEN_AI_ENDPOINT;
+import static com.azure.ai.openai.implementation.NonAzureOpenAIClientImpl.OPEN_AI_ENDPOINT;
+
+public class OpenAIClientBuilderTest {
+    @Test
+    public void testInnerImplBasedOnEndpoint() throws NoSuchFieldException, IllegalAccessException {
+        Field azureClient = OpenAIClient.class.getDeclaredField("serviceClient");
+        azureClient.setAccessible(true);
+        Field nonAzureClient = OpenAIClient.class.getDeclaredField("openAIServiceClient");
+        nonAzureClient.setAccessible(true);
```

Comment:
This is pretty brittle, but I'm not sure of a better way to test that the correct implementation is used. Maybe it's better to leave it un-unit-tested since it's all under the hood. Open to suggestions!